sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py

```diff
@@ -0,0 +1,138 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import logging
+import typing
+import unittest
+
+from past.builtins import unicode
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam.io.external.jdbc import WriteToJdbc
+from apache_beam.testing.test_pipeline import TestPipeline
+
+# pylint: disable=wrong-import-order, wrong-import-position, ungrouped-imports
+try:
+  import sqlalchemy
+except ImportError:
+  sqlalchemy = None
+# pylint: enable=wrong-import-order, wrong-import-position, ungrouped-imports
+
+# pylint: disable=wrong-import-order, wrong-import-position, ungrouped-imports
+try:
+  from testcontainers.postgres import PostgresContainer
+except ImportError:
+  PostgresContainer = None
+# pylint: enable=wrong-import-order, wrong-import-position, ungrouped-imports
+
+JdbcTestRow = typing.NamedTuple(
+    "JdbcTestRow",
+    [
+        ("f_id", int),
+        ("f_real", float),
+        ("f_string", unicode),
+    ],
+)
+
+coders.registry.register_coder(JdbcTestRow, coders.RowCoder)
+
+
+@unittest.skipIf(sqlalchemy is None, "sql alchemy package is not installed.")
+@unittest.skipIf(
+    PostgresContainer is None, "testcontainers package is not installed")
```

Comment:
I think we should add these packages to the test extra: https://github.com/apache/beam/blob/92170e85aa6284ad3908656ecc2d6f31505d0310/sdks/python/setup.py#L181-L198

cc: @aaltay in case there are any concerns with adding these dependencies. For licenses they're both under Apache 2.0 (and it's also just a test dependency).

Reply:
Thank you. I agree. It is fine to add as test only dependencies.

Reply:
Is it ok if the tests will run for python >=3.5 only? testcontainers don't support python2, as well as pytest-postgresql.

Reply:
I think that's fine, you can just add a skipIf based on the python version

Reply:
I left those skipIfs and added testcontainers only for python3 ```testcontainers<3.0.3; python_version >= "3.5"'```